textalyze.js

```diff
@@ -1,23 +1,30 @@
-/**
- * This is the base code for v0.1 of our JavaScript text analyzer.
- * Visit https://github.com/jfarmer/text-analysis to see what to do.
- *
- * Send a message in Slack if you're stuck or unsure what to do.  These
- * comments are here to help you, but please delete them as you go along.
- */
-
 /**
  * Given an input Array, returns a Map containing the count of each item in the input.
  * @param {Array} array - The array of items to count
  * @param {Map} counts - A Map containing the counts of the items in the input array
  */
 function itemCounts(array) {
-  let counts = new Map();
+  const counts = new Map();
 
-  // Your code here.
-  // Run 'npm test' to see what tests need to pass.
+  array.forEach((value) => {
+    const count = counts.get(value) || 0;
+    counts.set(value, count + 1);
+  });
 
   return counts;
 }
 
 module.exports = { itemCounts };
+
+
+//
+// running the app
+//
+
+const array = ['a', 'b', 'b', 'c', 'c', 'a', 'd'];
+
+console.log(`The counts for ${array} are...`);
+
+itemCounts(array).forEach((value, key) => {
+  console.log(`${key}\t${value}`);
```

Comment:
Check out [console.table](https://developer.mozilla.org/en-US/docs/Web/API/Console/table).   The output looks like this on the command-line: https://cl.ly/3bf339b0bd5d

Reply:
actually I disliked the index being printed